r2_chatterbot/main.py

```diff
@@ -30,38 +31,57 @@
 
 utils.set_classpath()
 
+
 def main():
+    print("Hello! I am C1C0. I can answer questions and execute commands.")
     while True:
-        #gets a tuple of phrase and confidence
+        # gets a tuple of phrase and confidence
         answer = live_streaming.main()
         speech = live_streaming.get_string(answer)
         confidence = live_streaming.get_confidence(answer)
         print(speech)
+
         if "quit" in speech or "stop" in speech:
             break
-        
+
         if("cico" in speech.lower() or "kiko" in speech.lower() or "c1c0" in speech.lower()) and \
                 ("hey" in speech.lower()):
             # filter out cico since it messes with location detection
             speech = utils.filter_cico(speech)
-            
+
             if face_recognition.isFaceRecognition(speech):
-                print(face_recognition.faceRecog(speech)) 
+                print(face_recognition.faceRecog(speech))
                 # task is to transfer over to facial recognition client program
             elif path_planning.isLocCommand(speech.lower()):
                 print("Move command (itemMove, direction, moveAmmount): ")
                 print(path_planning.pathPlanning(speech.lower()))
                 # task is to transfer over to path planning on the system
             elif object_detection.isObjCommand(speech.lower()):
-                print("Object to pick up: " + object_detection.object_parse(speech.lower()))
+                print("Object to pick up: " +
+                      object_detection.object_parse(speech.lower()))
                 # task is to transfer over to object detection on the system
             else:
                 # we don't want the text to be lowercase since it messes with location detection
-                print(corpus_and_adapter.response_from_chatbot(speech))
+                response = "Sorry, I don't understand."
+                data = keywords.get_topic(speech, parse_location=False)
+                keywords.modify_topic_data(data, parse_location=True)
+                if "name" in data.keys() and data["name"] == "weather":
+                    api_data = weather.lookup_weather_today_city(
+                        data["info"]["location"]["name"])
+                    response = make_response.make_response_api(data, api_data)
+                elif "name" in data.keys() and data["name"] == "restaurant":
+                    api_data = restaurant.lookup_restaurant_city(
+                        data["info"]["location"]["name"])
+                    response = make_response.make_response_api(data, api_data)
+                else:
+                    # Q&A system
+                    response = get_topic(speech)
+                print(response)
```

Comment:
Since chatterbot was also being used for weather/restaurant API questions, I added the functionality to answer these types of questions using the other util files. The speech is first checked to see if it contains weather or restaurant keywords, then it gets a response from the topic classification program.